model.py

```diff
@@ -0,0 +1,118 @@
+rolling_median_feature = pd.DataFrame()
+rolling_std_feature = pd.DataFrame()
+differential_feature = pd.DataFrame()
+
+# Shift() function in pandas creates these shifted or lag features from time series data.
+# Adding a lag of 1 day
+for lag in range(1,2):
+    for location in ['Alamosa','Grand','Springs','Denver']:
+        lag_featues[f"{location}_lag_{lag}"] = all_data[location].shift(lag)
+
+# Trying a few different rolling window sizes
+window_sizes = [1,3,5,7]
+
+for window_size in window_sizes:
+    for location in ['Alamosa','Grand','Springs','Denver']:
+        rolling_mean_feature[f"{location}_rolling_mean_{window_size}"] = all_data[location].rolling(window=window_size).mean()
+        rolling_std_feature[f"{location}_rolling_std_{window_size}"] = all_data[location].rolling(window=window_size).std()
+        rolling_median_feature[f"{location}_rolling_median_{window_size}"] = all_data[location].rolling(window=window_size).median()
+
+# Differential features
+for location in ['Alamosa','Grand','Springs','Denver']:
+    differential_feature[f"{location}_diff"] = all_data[location].diff()
+
+# Seasonality features
+all_data['day_of_year'] = all_data['DATE'].dt.dayofyear
+all_data['month'] = all_data['DATE'].dt.month
+
+all_data = pd.concat([all_data,lag_featues,rolling_mean_feature,rolling_median_feature,rolling_std_feature,differential_feature],axis=1)
+
+# print(all_data)
+'''
+           DATE    Alamosa     Denver  ...  Grand_diff  Springs_diff  Denver_diff
+0    2003-01-01  24.985507  33.523256  ...         NaN           NaN          NaN
+1    2003-01-02  15.527778  37.068966  ...    0.743245     -0.083118     3.545710
+2    2003-01-03  22.891156  47.344828  ...    1.414966     12.130359    10.275862
+3    2003-01-04  26.864583  39.812500  ...    2.224490     -5.221088    -7.532328
+4    2003-01-05  30.284722  41.914286  ...    1.248699      1.367551     2.101786
+...         ...        ...        ...  ...         ...           ...          ...
+7300 2022-12-27  31.413793  48.848485  ...   -1.017241     16.303571    12.385985
+7301 2022-12-28  36.068966  39.130435  ...    1.466292     -8.203571    -9.718050
+7302 2022-12-29  21.800000  25.578947  ...   -7.973539    -12.451351   -13.551487
+7303 2022-12-30  20.327586  26.424242  ...    4.146135     -0.683131     0.845295
+7304 2022-12-31  30.800000  32.794118  ...    3.043651     16.373192     6.369875
+'''
+
+# Handling missing values from lag and rolling features
+all_data.interpolate(method='linear',inplace=True)  
+all_data.fillna(method='bfill',inplace=True)
+all_data.fillna(method='ffill',inplace=True)
+
+print("NaNs after processing:")
+nan_counts = all_data.isna().sum()
+print(nan_counts)
+
+# Handling persistent NaNs
+if nan_counts.sum() > 0:
+    # drop columns with NaNs
+    all_data.dropna(axis=1, how='all', inplace=True)
+    # drop rows with NaNs
+    all_data.dropna(axis=0, how='any', inplace=True)
+
+
+model= RandomForestRegressor(n_estimators=100, random_state=42)

```

Comment:
Why did you choose `random_state=42`?

https://news.mit.edu/2019/answer-life-universe-and-everything-sum-three-cubes-mathematics-0910

Reply:
No particular reason, this seems to be common practice when working with machine learning models. I actually think that we should test different random states to see if how it would affect the performance